homeassistant/components/sensor/skybeacon.py

```diff
@@ -0,0 +1,181 @@
+    mon.start()
+
+
+class SkybeaconHumid(Entity):
+    """Representation of a humidity sensor."""
+
+    def __init__(self, name, mon):
+        """Initialize a sensor."""
+        self.mon = mon
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self.mon.humid
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return "%"
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the sensor."""
+        return {
+            ATTR_DEVICE: "SKYBEACON",
+            ATTR_MODEL: 1,
+        }
+
+class SkybeaconTemp(Entity):
+    """Representation of a temperature sensor."""
+
+    def __init__(self, name, mon):
+        """Initialize a sensor."""
+        self.mon = mon
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self.mon.temp
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return TEMP_CELSIUS
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the sensor."""
+        return {
+            ATTR_DEVICE: "SKYBEACON",
+            ATTR_MODEL: 1,
+        }
+
+class Monitor(threading.Thread):
+    """Connection handling."""
+
+    def __init__(self, hass, mac, name):
+        """Construct interface object."""
+        threading.Thread.__init__(self)
+        self.daemon = True
+        self.hass = hass
+        self.mac = mac
+        self.name = name
+        self.temp = STATE_UNKNOWN
+        self.humid = STATE_UNKNOWN
+
+    def run(self):
+        """Thread that keeps connection alive."""
+        adapter = pygatt.backends.GATTToolBackend()
+        while True:
+            try:
+                _LOGGER.info("connecting to %s", self.name)
+                # we need concurrent connect, so lets not reset the device
+                adapter.start(reset_on_start=False)
+                # seems only one connection can be initiated at a time
+                with CONNECT_LOCK:
+                    device = adapter.connect(self.mac,
+                                             CONNECT_TIMEOUT,
+                                             pygatt.BLEAddressType.random)
+                if SKIP_HANDLE_LOOKUP:
+                    # HACK: inject handle mapping collected offline
+                    # pylint: disable=protected-access
+                    device._characteristics[UUID(BLE_TEMP_UUID)] = CACHED_CHR
+                # magic: writing this makes device happy
+                device.char_write_handle(0x1b, bytearray([255]), False)
```

Comment:
This kind of code indicates that this logic (and thus class) should be extracted into a standalone lib.